### PR TITLE
Allow adding extra keys to Info.plist

### LINKED ISSUE #795: Support for custom Info.plist entries
For setting up deeplinking on macOS using `Desktop.getDesktop.setOpenURIHandler()`, Apple requires these entries to be present in its `Info.plist` file:

```xml
<key>CFBundleURLTypes</key>
<array>
  <dict>
    <key>CFBundleURLName</key>
    <string>PACKAGE NAME OF APP</string>
    <key>CFBundleURLSchemes</key>
    <array>
      <string>DEEPLINK SCHEME</string>
    </array>
  </dict>
</array>
```

It'd be nice if compose for desktop offers a standard API for contributing custom entries to `Info.plist`. This is possibly related to https://github.com/JetBrains/compose-jb/issues/773.

### PR DESCRIPTION
**Pull request by AlexeyTsvetkov**
Resolves #795

### CODE REVIEW

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/InfoPlistBuilder.kt [hunk 1]
```diff
@@ -5,10 +5,11 @@
 
 package org.jetbrains.compose.desktop.application.internal
 
+import org.jetbrains.compose.desktop.application.dsl.InfoPlistSettings
 import java.io.File
 import kotlin.reflect.KProperty
 
-internal class InfoPlistBuilder {
+internal class InfoPlistBuilder(private val extraPlistKeysRawXml: String?) {
```
Comment (olonho) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/InfoPlistBuilder.kt:12]: `private val extraPlistKeysRawXml: String? = null`?

File: tutorials/Native_distributions_and_local_execution/README.md [hunk 2]
```diff
@@ -408,3 +409,79 @@ compose.desktop {
...
+                    extraKeysRawXml = macExtraPlistKeys
+                }
+            }
+        }
+    }
+}
+
+val macExtraPlistKeys: String
+    get() = """
+      <key>CFBundleURLTypes</key>
```
Comment (olonho) [inline tutorials/Native_distributions_and_local_execution/README.md:443]: Docs at https://docs.oracle.com/en/java/javase/12/docs/api/java.desktop/java/awt/Desktop.html#setOpenURIHandler(java.awt.desktop.OpenURIHandler) says that it only works when `CFBundleDocumentTypes` is present. Is it actually correct?

File: tutorials/Native_distributions_and_local_execution/README.md [hunk 3]
```diff
@@ -408,3 +409,79 @@ compose.desktop {
...
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import java.awt.Desktop
+
+fun main() {
+    val text = mutableStateOf("Hello, World!")
+
+    Desktop.getDesktop().setOpenURIHandler { event ->
+        text.value = "Open URI: " + event.getURI()
```
Comment (olonho) [inline tutorials/Native_distributions_and_local_execution/README.md:475]: Maybe `event.uri`?

File: tutorials/Native_distributions_and_local_execution/README.md [hunk 4]
```diff
@@ -408,3 +409,79 @@ compose.desktop {
...
+
+val macExtraPlistKeys: String
+    get() = """
+      <key>CFBundleURLTypes</key>
+      <array>
+        <dict>
+          <key>CFBundleURLName</key>
+          <string>Example deep link</string>
+          <key>CFBundleURLSchemes</key>
+          <array>
```
Comment (olonho) [inline tutorials/Native_distributions_and_local_execution/README.md:449]: How one defines supported scheme for other platforms, i.e. Windows or Linux?
Per https://stackoverflow.com/questions/50788945/how-to-make-windows-java-based-app-handle-deep-linking seems that we need to patch manifest or registry.

We could merge this change, as it fixes macOS, but could think of other platforms support.

**Review by olonho — APPROVED** [review]
Overall looks good, few minor nits.

File: tutorials/Native_distributions_and_local_execution/README.md (hunk 2 above)
Comment (AlexeyTsvetkov) [reply tutorials/Native_distributions_and_local_execution/README.md:443]: Does not seem so (I've checked the example on Big Sur and it worked)

File: tutorials/Native_distributions_and_local_execution/README.md (hunk 4 above)
Comment (AlexeyTsvetkov) [reply tutorials/Native_distributions_and_local_execution/README.md:449]: I've created a separate issue https://github.com/JetBrains/compose-jb/issues/855

**Pull request merged**